src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -71,36 +71,31 @@ public void list_noop(ChocolateyConfiguration config)
                                 ));
         }
 
-        public ConcurrentDictionary<string, PackageResult> list_run(ChocolateyConfiguration config, bool logResults = true)
+        public IEnumerable<PackageResult> list_run(ChocolateyConfiguration config, bool logResults)
         {
-            var packageResults = new ConcurrentDictionary<string, PackageResult>();
-
-            var packages = NugetList.GetPackages(config, _nugetLogger).ToList();
-
-            foreach (var package in packages.or_empty_list_if_null())
+            foreach (var package in NugetList.GetPackages(config, _nugetLogger))
             {
+                var pkg = package; // for lamda access

```

Comment:
My thoughts about switching `package` to `pkg` on line 76 and making this lamdba-safe var named `package` still apply here.